src/flp/szk.rs

```diff
@@ -498,12 +501,14 @@ where
 
     pub(crate) fn query(
         &self,
+        ctx: &[u8],
+        level: u16, // level of the prefix tree
```

Comment:
This is per the spec. It's slightly awkward since a detail of VIDPF is bleeding into SZK. However, I think this feature of the spec is useful because it ensures independence of query randomness at different levels of the tree. This isn't strictly necessary given that the SZK is queried once, but is useful for defense in depth in case an aggregator inadvertently allows multiple queries.